internal/services/codesigning/trusted_signing_certificate_profile_resource.go

```diff
@@ -0,0 +1,478 @@
+						Computed: true,
+					},
+
+					"expiry_date": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"revocation": {
+						Type:     pluginsdk.TypeList,
+						Computed: true,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"effective_at": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+
+								"failure_reason": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+
+								"reason": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+
+								"remarks": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+
+								"requested_at": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+
+								"status": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+							},
+						},
+					},
+
+					"serial_number": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"status": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"subject_name": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"thumbprint": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+				},
+			},
+		},
+
+		"status": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (r TrustedSigningCertificateProfileResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.CodeSigning.Client.CertificateProfiles
+
+			var model TrustedSigningCertificateProfileModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			codeSigningAccountId, err := codesigningaccounts.ParseCodeSigningAccountID(model.TrustedSigningAccountId)
+			if err != nil {
+				return err
+			}
+
+			id := certificateprofiles.NewCertificateProfileID(codeSigningAccountId.SubscriptionId, codeSigningAccountId.ResourceGroupName, codeSigningAccountId.CodeSigningAccountName, model.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			properties := &certificateprofiles.CertificateProfile{
+				Properties: &certificateprofiles.CertificateProfileProperties{
+					IdentityValidationId: model.IdentityValidationId,
+					IncludeCity:          &model.IncludeCity,
+					IncludeCountry:       &model.IncludeCountry,
+					IncludePostalCode:    &model.IncludePostalCode,
+					IncludeState:         &model.IncludeState,
+					IncludeStreetAddress: &model.IncludeStreetAddress,
```

Comment:
Some of the attributes here *applicable only for private trust, private trust ci profile type", does API allow to pass even a false here for other profile type? Maybe we shall add a test case to cover this.

Reply:
we can not test this for now since we can not do public identity validation, but confirmed with service team, it can be set to `false` in other profile type. Does it make sense?